### PR TITLE
[CAM-7203] - Allow more that one job per GitHub on Jenkins plugin

### PR DESCRIPTION
**Pull request by julio-castanar**
@HectorGarciaAlvarez @gsanchezu @afavieres
Could you review this pull request, please?

### CODE REVIEW

File: src/main/java/com/elasticbox/jenkins/triggers/github/PullRequestBuildHandler.java [hunk 1]
```diff
@@ -433,14 +434,23 @@ private boolean isWhitelisted(GHUser user, GitHub gitHub) throws IOException {
...
+        String itemProjectName = null;
+        String itemRequestUrl = null;
         ParametersAction parameters = actionable.getAction(ParametersAction.class);
 
         if (parameters == null) {
             return false;
         }
         for (ParameterValue parameter: parameters.getParameters()) {
-            if ("PR_URL".equals(parameter.getName()) && pullRequestUrl.equals(parameter.getValue())) {
+            if ("PR_URL".equals(parameter.getName())) {
```
Comment (gsanchezu) [inline src/main/java/com/elasticbox/jenkins/triggers/github/PullRequestBuildHandler.java]: Please, use the constant defined above

File: src/main/java/com/elasticbox/jenkins/triggers/github/PullRequestBuildHandler.java [hunk 2]
```diff
@@ -433,14 +434,23 @@ private boolean isWhitelisted(GHUser user, GitHub gitHub) throws IOException {
...
+                itemRequestUrl = (String) parameter.getValue();
+            } else if ("PROJECT_NAME".equals(parameter.getName())) {
```
Comment (gsanchezu) [inline src/main/java/com/elasticbox/jenkins/triggers/github/PullRequestBuildHandler.java]: Please, use the constant defined above

File: src/main/java/com/elasticbox/jenkins/triggers/github/PullRequestBuildHandler.java [hunk 3]
```diff
@@ -433,14 +434,23 @@ private boolean isWhitelisted(GHUser user, GitHub gitHub) throws IOException {
...
+                itemProjectName = (String) parameter.getValue();
+            }
+
+            if ( (itemRequestUrl != null) && (itemProjectName != null)
```
Comment (gsanchezu) [inline src/main/java/com/elasticbox/jenkins/triggers/github/PullRequestBuildHandler.java]: If you are sure the parameters are not null, you can flip the equals objects and remove the null check.

Comment (afavieres) [inline src/main/java/com/elasticbox/jenkins/triggers/github/PullRequestBuildHandler.java]: Unless parameters.getparameters() is a slow call or the number of parameters is usually bigger than a few thousands I prefer something less clever.

```suggestion
            itemRequestUrl = extractRequestUrl(parameters);
            itemProjectName = extractProjectName(parameters);
```

File: src/main/java/com/elasticbox/jenkins/triggers/github/PullRequestBuildHandler.java (hunk 2 above)
Comment (afavieres) [inline src/main/java/com/elasticbox/jenkins/triggers/github/PullRequestBuildHandler.java]: Use the existing constant PROJECT_NAME.

**Comment by allyourbotsarebelongtous** [comment]
Deployed ElasticBox instance is available at https://10.166.246.72 and Jenkins instance at https://10.166.246.80

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/ebx_pull_request_jenkins_plugin/529/ for more details.

File: src/main/java/com/elasticbox/jenkins/triggers/github/PullRequestBuildHandler.java (hunk 3 above)
Comment (julio-castanar) [reply src/main/java/com/elasticbox/jenkins/triggers/github/PullRequestBuildHandler.java]: ok

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/ebx_pull_request_jenkins_plugin/530/ for more details.

File: src/main/java/com/elasticbox/jenkins/triggers/github/DeleteInstancesWorkload.java [hunk 4]
```diff
@@ -101,7 +101,8 @@ private boolean deleteInstance(PullRequestInstance instance, TaskListener listen
             } catch (ClientException ex) {
                 if (ex.getStatusCode() == HttpStatus.SC_CONFLICT) {
                     return false;
-                } else if (ex.getStatusCode() != HttpStatus.SC_NOT_FOUND) {
+                } else if ( (ex.getStatusCode() != HttpStatus.SC_FORBIDDEN)
+                        && (ex.getStatusCode() != HttpStatus.SC_NOT_FOUND) ) {
                     throw ex;
                 }
```
[the remark this answers is not in the document]
Comment (afavieres) [reply src/main/java/com/elasticbox/jenkins/triggers/github/DeleteInstancesWorkload.java]: This is not what the code is doing.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/ebx_pull_request_jenkins_plugin/531/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Deployed ElasticBox instance is available at https://10.166.246.72 and Jenkins instance at https://10.166.246.80

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/ebx_pull_request_jenkins_plugin/532/ for more details.

File: src/main/java/com/elasticbox/jenkins/triggers/github/PullRequestBuildHandler.java [hunk 5]
```diff
@@ -433,19 +434,23 @@ private boolean isWhitelisted(GHUser user, GitHub gitHub) throws IOException {
...
-                return true;
+            if (PR_URL.equals(parameter.getName())) {
+                itemRequestUrl = (String) parameter.getValue();
+            } else if (PROJECT_NAME.equals(parameter.getName())) {
+                itemProjectName = (String) parameter.getValue();
             }
         }
 
-        return false;
+        return ( (itemProjectName.equals(projectName)) && (itemRequestUrl.equals(pullRequestUrl)) ) ;
```
Comment (gsanchezu) [inline src/main/java/com/elasticbox/jenkins/triggers/github/PullRequestBuildHandler.java]: Please, flip the objects in the equals, to better ensure they are not null

Comment (gsanchezu) [reply src/main/java/com/elasticbox/jenkins/triggers/github/PullRequestBuildHandler.java]: Hey @julio-castanar, let's fix this in the next one, I want to test the release job today

**Comment by allyourbotsarebelongtous** [comment]
The following instances are being terminated: https://ops.elasticbox.com/services/instances/i-vdzpb0, https://ops.elasticbox.com/services/instances/i-3a2s2p

**Pull request merged**